p2p/pubsubManager/gossipsub.go

```diff
@@ -135,6 +135,16 @@ func (g *PubsubManager) Subscribe(location common.Location, datatype interface{}
 	return nil
 }
 
+// unsubscribe from broadcasts of the given type of data
+func (g *PubsubManager) Unsubscribe(location common.Location, datatype interface{}) {
+	if topicName, err := TopicName(g.genesis, location, datatype); err != nil {
+		g.subscriptions[topicName].Cancel()
```

Comment:
is this lock safe?

Reply:
What lock? Or do you mean "lock safe", in which case.. what do you mean by "lock safe"?